src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -112,6 +112,28 @@ struct LiftStringFunction {
   }
 };
 
+// IRI or URI
+//
+// TODOs:
+//
+// 1. Check for `BASE` URL and if it exists, prepend it.
+// 2. What's the correct behavior for non-strings, like `1` or `true`?
+//
+// @1: Ignored by the implementation below.
+// @2: The implementation below converts all this to a string and then to an
+// IRI.
+[[maybe_unused]] auto iriOrUri =
+    [](std::optional<std::string> s) -> IdOrLiteralOrIri {
+  if (!s.has_value()) {
+    return Id::makeUndefined();
+  } else {
+    return IdOrLiteralOrIri{
+        LiteralOrIri{Iri::fromIrirefWithoutBrackets(s.value())}};
+  }
```

Comment:
I thinkt the easiest way to implement this is to use a custom
ValueGetter that directly returns the result of this expression and make the actual function `std::identity`.
This is very often a good way to implement unary expressions.